File: competitive_programming/leetcode/01_python/28_find_index_first_occurrence.py
"""LeetCode 28 - Find the Index of the First Occurrence in a String."""
def str_str(haystack, needle):
    return haystack.find(needle)


if __name__ == "__main__":
    assert str_str("sadbutsad", "sad") == 0
    assert str_str("leetcode", "leeto") == -1
    print("28 find index first occurrence ok")

File: competitive_programming/leetcode/03_cpp/28_find_index_first_occurrence.cpp
#include <algorithm>
#include <cassert>
#include <cmath>
#include <cstdint>
#include <iostream>
#include <set>
#include <string>
#include <vector>

int strStr(const std::string &haystack, const std::string &needle) {
    auto position = haystack.find(needle);
    return position == std::string::npos ? -1 : static_cast<int>(position);
}

int main() {
    assert(strStr("sadbutsad", "sad") == 0 && strStr("leetcode", "leeto") == -1);
    std::cout << "28 find index first occurrence ok\n";
    return 0;
}
